shared/js/smart-screen/card_store.js

```diff
@@ -104,10 +103,50 @@
           resolve();
         }
       });
+    },
+
+    iterateData: function cs_listData(cb) {
+      return new Promise((resolve, reject) => {
+
+        this._getStore().then(store => {
+          return store.sync();
+
+        }).then(function iterateTask(cursor) {
+          cursor.next().then(task => {
```

Comment:
I suggest following the sample code in [1] to avoid potential risks.

[1] https://developer.mozilla.org/en-US/docs/Web/API/DataStore/sync#Example